proposed/2022/NuGetExeWarnWhenSDKProjectsGetPacked.md

```diff
@@ -0,0 +1,97 @@
+It is recommended that SDK projects are managed by using `dotnet.exe` instead.
+
+## Explanation
+
+### Functional explanation
+
+#### How packing and SDK-based project with NuGet.exe behaves today
+
+Scenario 1: Single target framework SDK based projects, with .NET SDK installed on the machine.
+
+```console
+Attempting to build package from 'SDK.csproj'.
+MSBuild auto-detection: using msbuild version '17.4.0.37601' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64'.
+Packing files from 'C:\Code\Temp\SDK\bin\Debug\net6.0'.
+WARNING: NU5115: Description was not specified. Using 'Description'.
+WARNING: NU5115: Author was not specified. Using 'Roki2'.
+WARNING: NU5128: Some target frameworks declared in the dependencies group of the nuspec and the lib/ref folder do not have exact matches in the other location. Consult the list of actions below:
+- Add a dependency group for net6.0 to the nuspec
+Successfully created package 'C:\Code\Temp\SDK\SDK.1.0.0.nupkg'.
+```
+
+The command packages the single assembly, but does not generate the correct dependencies or any other metadata in the same way that the pack SDK would.
+
+Scenario 2: Multiple target framework SDK based projects, with .NET SDK installed on the machine.
+
+```console
+Attempting to build package from 'SDK.csproj'.
+MSBuild auto-detection: using msbuild version '17.4.0.37601' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64'.
+Error NU5012: Unable to find 'bin\Debug\SDK\bin\Debug\'. Make sure the project has been built.
+```
+
+NuGet.exe does not understand the concept of multitargetting and just merges the output paths.
+
+Scenario 3: SDK-based project, without the .NET SDK installed
+
+```console
+The SDK 'Microsoft.NET.Sdk' specified could not be found.  C:\Code\Temp\SDK\SDK.csproj
+```
+
+The parsing fails early.
+
+### Proposal
+
+When someone attempts to run pack on an SDK-based project, NuGet will fail with the following error:
+
+```console
+Error NU5049: NuGet.exe does not support packing SDK-based projects, use `dotnet.exe pack` to pack this project instead.
+```
+
+### Technical explanation
+
+As described above, whenever there's no SDK installed, there's simply no accurate way for NuGet to detect that a project is SDK-based.
+We can do some XML parsing, but it is likely to be error prone. `NuGet.exe pack` is not the only command that will fail, so the customer will likely figure out that they need to install the .NET SDK. After that the same scenarios with the installed .NET SDK will follow.
+
+To determine whether a project is `.NET SDK` based, we will evaluate the project (part of the normal pack operation), and check for the `UsingMicrosoftNETSDK` property being set. If the property is set, then the project is presumed to be .NET SDK based and will be blocked from packing.

```

Comment:
Should we check `PackageRestoreStyle` == `PackageReference`, so that legacy csproj PackageReference projects also get captured?

Reply:
I think it'd be easy to add that too. I have it in `Future Possiblities` right now. 
I'm pretty split on it, curious what others think.

Reply:
Does NuGet.exe pack work today for legacy csproj `PackageReference` projects? What will be the value of `UsingMicrosoftNETSDK` property for legacy `PackageReference` projects? If the value is `No` then I would assume the current implementation will raise an error. Is my understanding correct?

Reply:
> Does NuGet.exe pack work today for legacy csproj PackageReference projects

It does not. `NuGet.exe in general won't work well with PackageReference projects`. 

> What will be the value of UsingMicrosoftNETSDK property for legacy PackageReference projects?

It will be empty. csproj PackageReference projects *do not* have the pack targets by default, so you'd need to install a package in order to make the project packable. 

NuGet.exe will complete the packing, pack the assembly, but the metadata and dependencies will not be correct.